cmk/base/plugins/agent_based/printer_supply.py

```diff
@@ -132,7 +132,8 @@ def check_printer_supply(item: str, params: Mapping[str, Any], section: Section)
 
     color_info = ""
     if supply.color and supply.color.lower() not in item.lower():
-        color_info = "[%s] " % supply.color
+        # fix trailing zero byte in output (seen on HP Jetdirect 153)
+        color_info = "[%s] " % supply.color.rstrip('\0')
```

Comment:
The fix itself looks ok, but this is the wrong place, we should fix this in the parse function:
`parsed[description] = PrinterSupply(unit, max_capacity, level, supply_class, color.rstrip('\0'))`

Futhermore, please add a corresponding unit test to tests/unit/cmk/base/plugins/agent_based/test_printer_supply.py:
```
def test_parse_printer_supply():
    assert parse_printer_supply([input from you walk]) == [expected result without any null bytes]
```

Reply:
Thank you for the feedback! Just one question: As there already is a line regarding stripping variables of zero bytes in the parse-function, would it be more clear if we put it in an extra line after that, like so:
```python
.....
89         # fix trailing zero bytes (seen on HP Jetdirect 143 and 153)
90         description = name.split(" S/N:")[0].strip("\0")
91         color = color.rstrip("\0")
92         unit = get_unit(unit_info)
....
```
I don't know if an assignment like that is compliant with your coding-style. If not, i can just put the `rstrip` in the line you suggested.

Reply:
That would be fine and compliant with our coding-style, its probably more readable that what I suggested.

Reply:
I just pushed the change where i moved the sanitization to the parse-function.
I also added a unit test with the part of the walk that was causing the problems as input data. I tried to adapt the structure of the unit test to the ones that where already there, but if you had something else in mind for the test just let me know.

Reply:
Thanks for the changes, looks good now.